### ISSUE TITLE
building shared library libcrypto.so.43 failed

### ISSUE THREAD

**Issue by ericbsd** [trueos/trueos#30]
```
building shared library libcrypto.so.43
cc -target x86_64-unknown-freebsd12.0 --sysroot=/usr/obj/usr/home/eturgeon/projects/trueos/trueos/amd64.amd64/tmp -B/usr/obj/usr/home/eturgeon/projects/trueos/trueos/amd64.amd64/tmp/usr/bin   -fstack-protector-strong -shared -Wl,-x -Wl,--fatal-warnings -Wl,--warn-shared-textrel  -o libcrypto.so.43.full -Wl,-soname,libcrypto.so.43  `NM='nm' NMFLAGS='' lorder freezero.pico getentropy_freebsd.pico recallocarray.pico timingsafe_bcmp.pico timingsafe_memcmp.pico cryptlib.pico malloc-wrapper.pico mem_dbg.pico cversion.pico ex_data.pico cpt_err.pico o_time.pico o_str.pico o_init.pico mem_clr.pico crypto_init.pico aes_misc.pico aes_ecb.pico aes_cfb.pico aes_ofb.pico aes_ctr.pico aes_ige.pico aes_wrap.pico a_object.pico a_bitstr.pico a_time.pico a_int.pico a_octet.pico a_print.pico a_type.pico a_dup.pico a_d2i_fp.pico a_i2d_fp.pico a_enum.pico a_utf8.pico a_sign.pico a_digest.pico a_verify.pico a_mbstr.pico a_strex.pico x_algor.pico x_val.pico x_pubkey.pico x_sig.pico x_req.pico x_attrib.pico x_bignum.pico x_long.pico x_name.pico x_x509.pico x_x509a.pico x_crl.pico x_info.pico x_spki.pico nsseq.pico x_nx509.pico d2i_pu.pico d2i_pr.pico i2d_pu.pico i2d_pr.pico t_req.pico t_x509.pico t_x509a.pico t_crl.pico t_pkey.pico t_spki.pico t_bitst.pico tasn_new.pico tasn_fre.pico tasn_enc.pico tasn_dec.pico tasn_utl.pico tasn_typ.pico tasn_prn.pico ameth_lib.pico f_int.pico f_string.pico n_pkey.pico f_enum.pico x_pkey.pico a_bool.pico x_exten.pico bio_asn1.pico bio_ndef.pico asn_mime.pico asn1_gen.pico asn1_par.pico asn1_lib.pico asn1_err.pico a_bytes.pico a_strnid.pico evp_asn1.pico asn_pack.pico p5_pbe.pico p5_pbev2.pico p8_pkey.pico asn_moid.pico a_set.pico a_time_tm.pico bf_skey.pico bf_ecb.pico bf_cfb64.pico bf_ofb64.pico bio_lib.pico bio_cb.pico bio_err.pico bio_meth.pico bss_mem.pico bss_null.pico bss_fd.pico bss_file.pico bss_sock.pico bss_conn.pico bf_null.pico bf_buff.pico b_print.pico b_dump.pico b_posix.pico b_sock.pico bss_acpt.pico bf_nbio.pico bss_log.pico bss_bio.pico bss_dgram.pico bn_add.pico bn_div.pico bn_exp.pico bn_lib.pico bn_ctx.pico bn_mul.pico bn_mod.pico bn_print.pico bn_rand.pico bn_shift.pico bn_word.pico bn_blind.pico bn_kron.pico bn_sqrt.pico bn_gcd.pico bn_prime.pico bn_err.pico bn_sqr.pico bn_recp.pico bn_mont.pico bn_mpi.pico bn_exp2.pico bn_gf2m.pico bn_nist.pico bn_depr.pico bn_const.pico bn_x931p.pico buffer.pico buf_err.pico buf_str.pico cmll_cfb.pico cmll_ctr.pico cmll_ecb.pico cmll_ofb.pico c_skey.pico c_ecb.pico c_enc.pico c_cfb64.pico c_ofb64.pico chacha.pico cmac.pico cm_ameth.pico cm_pmeth.pico comp_lib.pico comp_err.pico c_rle.pico c_zlib.pico conf_err.pico conf_lib.pico conf_api.pico conf_def.pico conf_mod.pico conf_mall.pico conf_sap.pico curve25519.pico curve25519-generic.pico cbc_cksm.pico cbc_enc.pico cfb64enc.pico cfb_enc.pico ecb3_enc.pico ecb_enc.pico enc_read.pico enc_writ.pico fcrypt.pico ofb64enc.pico ofb_enc.pico pcbc_enc.pico qud_cksm.pico rand_key.pico set_key.pico xcbc_enc.pico str2key.pico cfb64ede.pico ofb64ede.pico ede_cbcm_enc.pico dh_asn1.pico dh_gen.pico dh_key.pico dh_lib.pico dh_check.pico dh_err.pico dh_depr.pico dh_ameth.pico dh_pmeth.pico dh_prn.pico dsa_gen.pico dsa_key.pico dsa_lib.pico dsa_asn1.pico dsa_vrf.pico dsa_sign.pico dsa_err.pico dsa_ossl.pico dsa_depr.pico dsa_ameth.pico dsa_pmeth.pico dsa_prn.pico dsa_meth.pico dso_dlfcn.pico dso_err.pico dso_lib.pico dso_null.pico dso_openssl.pico ec_lib.pico ecp_smpl.pico ecp_mont.pico ecp_nist.pico ec_cvt.pico ec_mult.pico ec_err.pico ec_curve.pico ec_check.pico ec_print.pico ec_asn1.pico ec_key.pico ec2_smpl.pico ec2_mult.pico ec_ameth.pico ec_pmeth.pico eck_prn.pico ecp_nist.pico ecp_oct.pico ec2_oct.pico ec_oct.pico ech_lib.pico ech_key.pico ech_err.pico ecs_lib.pico ecs_asn1.pico ecs_ossl.pico ecs_sign.pico ecs_vrf.pico ecs_err.pico eng_err.pico eng_lib.pico eng_list.pico eng_init.pico eng_ctrl.pico eng_table.pico eng_pkey.pico eng_fat.pico eng_all.pico tb_rsa.pico tb_dsa.pico tb_ecdsa.pico tb_dh.pico tb_ecdh.pico tb_rand.pico tb_store.pico tb_cipher.pico tb_digest.pico tb_pkmeth.pico tb_asnmth.pico eng_openssl.pico eng_cnf.pico eng_dyn.pico err.pico err_all.pico err_prn.pico encode.pico digest.pico evp_enc.pico evp_key.pico e_des.pico e_bf.pico e_idea.pico e_des3.pico e_camellia.pico e_rc4.pico e_aes.pico names.pico e_xcbc_d.pico e_rc2.pico e_cast.pico m_null.pico m_md4.pico m_md5.pico m_sha1.pico m_wp.pico m_dss.pico m_dss1.pico m_ripemd.pico m_ecdsa.pico p_open.pico p_seal.pico p_sign.pico p_verify.pico p_lib.pico p_enc.pico p_dec.pico bio_md.pico bio_b64.pico bio_enc.pico evp_err.pico e_null.pico c_all.pico evp_lib.pico evp_pkey.pico evp_pbe.pico p5_crpt.pico p5_crpt2.pico e_old.pico pmeth_lib.pico pmeth_fn.pico pmeth_gn.pico m_sigver.pico e_aes_cbc_hmac_sha1.pico e_rc4_hmac_md5.pico e_chacha.pico evp_aead.pico e_chacha20poly1305.pico e_gost2814789.pico m_gost2814789.pico m_gostr341194.pico m_streebog.pico m_md5_sha1.pico gost2814789.pico gost89_keywrap.pico gost89_params.pico gost89imit_ameth.pico gost89imit_pmeth.pico gost_asn1.pico gost_err.pico gostr341001.pico gostr341001_ameth.pico gostr341001_key.pico gostr341001_params.pico gostr341001_pmeth.pico gostr341194.pico streebog.pico hkdf.pico hmac.pico hm_ameth.pico hm_pmeth.pico i_cbc.pico i_cfb64.pico i_ofb64.pico i_ecb.pico i_skey.pico lhash.pico lh_stats.pico md4_dgst.pico md4_one.pico md5_dgst.pico md5_one.pico cbc128.pico ctr128.pico cts128.pico cfb128.pico ofb128.pico gcm128.pico ccm128.pico xts128.pico o_names.pico obj_dat.pico obj_lib.pico obj_err.pico obj_xref.pico ocsp_asn.pico ocsp_ext.pico ocsp_ht.pico ocsp_lib.pico ocsp_cl.pico ocsp_srv.pico ocsp_prn.pico ocsp_vfy.pico ocsp_err.pico pem_sign.pico pem_seal.pico pem_info.pico pem_lib.pico pem_all.pico pem_err.pico pem_x509.pico pem_xaux.pico pem_oth.pico pem_pk8.pico pem_pkey.pico pvkfmt.pico p12_add.pico p12_asn.pico p12_attr.pico p12_crpt.pico p12_crt.pico p12_decr.pico p12_init.pico p12_key.pico p12_kiss.pico p12_mutl.pico p12_utl.pico p12_npas.pico pk12err.pico p12_p8d.pico p12_p8e.pico pk7_asn1.pico pk7_lib.pico pkcs7err.pico pk7_doit.pico pk7_smime.pico pk7_attr.pico pk7_mime.pico bio_pk7.pico poly1305.pico randfile.pico rand_lib.pico rand_err.pico rc2_ecb.pico rc2_skey.pico rc2_cbc.pico rc2cfb64.pico rc2ofb64.pico rmd_dgst.pico rmd_one.pico rsa_eay.pico rsa_gen.pico rsa_lib.pico rsa_sign.pico rsa_saos.pico rsa_err.pico rsa_pk1.pico rsa_none.pico rsa_oaep.pico rsa_chk.pico rsa_pss.pico rsa_x931.pico rsa_asn1.pico rsa_depr.pico rsa_ameth.pico rsa_prn.pico rsa_pmeth.pico rsa_crpt.pico rsa_meth.pico sha1dgst.pico sha1_one.pico sha256.pico sha512.pico stack.pico ts_err.pico ts_req_utils.pico ts_req_print.pico ts_rsp_utils.pico ts_rsp_print.pico ts_rsp_sign.pico ts_rsp_verify.pico ts_verify_ctx.pico ts_lib.pico ts_conf.pico ts_asn1.pico txt_db.pico ui_err.pico ui_lib.pico ui_openssl.pico ui_util.pico wp_dgst.pico x509_def.pico x509_d2.pico x509_r2x.pico x509_cmp.pico x509_obj.pico x509_req.pico x509spki.pico x509_vfy.pico x509_set.pico x509cset.pico x509rset.pico x509_err.pico x509name.pico x509_v3.pico x509_ext.pico x509_att.pico x509type.pico x509_lu.pico x_all.pico x509_txt.pico x509_trs.pico by_file.pico by_dir.pico by_mem.pico x509_vpm.pico v3_bcons.pico v3_bitst.pico v3_conf.pico v3_extku.pico v3_ia5.pico v3_lib.pico v3_prn.pico v3_utl.pico v3err.pico v3_genn.pico v3_alt.pico v3_skey.pico v3_akey.pico v3_pku.pico v3_int.pico v3_enum.pico v3_sxnet.pico v3_cpols.pico v3_crld.pico v3_purp.pico v3_info.pico v3_ocsp.pico v3_akeya.pico v3_pmaps.pico v3_pcons.pico v3_ncons.pico v3_pcia.pico v3_pci.pico pcy_cache.pico pcy_node.pico pcy_data.pico pcy_map.pico pcy_tree.pico pcy_lib.pico aes-elf-x86_64.pico bsaes-elf-x86_64.pico vpaes-elf-x86_64.pico aesni-elf-x86_64.pico aesni-sha1-elf-x86_64.pico bn_asm.pico bf_enc.pico modexp512-elf-x86_64.pico mont-elf-x86_64.pico mont5-elf-x86_64.pico gf2m-elf-x86_64.pico cmll_misc.pico cmll-elf-x86_64.pico des_enc.pico fcrypt_b.pico md5-elf-x86_64.pico ghash-elf-x86_64.pico rc4-elf-x86_64.pico rc4-md5-elf-x86_64.pico sha1-elf-x86_64.pico sha256-elf-x86_64.pico sha512-elf-x86_64.pico wp-elf-x86_64.pico wp-macosx-x86_64.pico cpuid-elf-x86_64.pico |  tsort -q` -lthr
/usr/obj/usr/home/eturgeon/projects/trueos/trueos/amd64.amd64/tmp/usr/bin/ld: error: unable to find library -lthr
cc: error: linker command failed with exit code 1 (use -v to see invocation)
*** Error code 1

Stop.
make[4]: stopped in /usr/home/eturgeon/projects/trueos/trueos/secure/lib/libcrypto
*** Error code 1

Stop.
make[3]: stopped in /usr/home/eturgeon/projects/trueos/trueos
*** Error code 1

Stop.
make[2]: stopped in /usr/home/eturgeon/projects/trueos/trueos
*** Error code 1

Stop.
make[1]: stopped in /usr/home/eturgeon/projects/trueos/trueos
*** Error code 1

Stop.
make: stopped in /usr/home/eturgeon/projects/trueos/trueos
eturgeon@eturgeon ~/p/t/trueos> sudo make
Password:

Explicit target required.  Likely "buildworld" is wanted.  See build(7).

*** Error code 1

Stop.
make: stopped in /usr/home/eturgeon/projects/trueos/trueos
```

**Comment by kmoore134** [comment]
Bernard, I'm thinking this is related to your adding of -lthr to the libcrypto build?

**Comment by Sp1l** [comment]
@kmoore134 It is!

It's already been fixed with the latest commit, see freebsd/freebsd-ports@f06b73c867fb33776df505bc9f061ec473f08f34

**Comment by kmoore134** [comment]
@Sp1l What is the fix then for world, to also switch it to -pthread? Do you want to make the fix or shall I?

**Comment by Sp1l** [comment]
Was shooting from the hip... Base wasn't adapted indeed!

Fixed by f047f0c82db1

**Comment by ericbsd** [comment]
It now works for me.
